### ISSUE TITLE
batch tests sometimes fail due to "database not found:" errors

### ISSUE THREAD

**Issue by DreadPirateShawn** [gpestana/kapacitor-unit#43]
Sometimes batch tests can fail, citing "database not found:".

Sample run:
```
I0326 03:30:26.170480     147 kapacitor.go:33] DEBUG:: Kapacitor loading task: alert_weather_batch.tick
I0326 03:30:26.170561     147 kapacitor.go:42] DEBUG:: batch script after replace: 
var weather = batch
	| query('''
		SELECT mean(temperature)
		FROM "weather"."default"."temperature"
		''')
			.period(5m)
			.every(1s)

var snow = batch
	| query('''
		SELECT count(precipitation)
		FROM "weather"."default"."snow"
	''')
		.period(3h)
		.every(1s)
		.groupBy(time(1h), 'location')
		.alignGroup()


// simple case with only one batch query
//   https://github.com/gpestana/kapacitor-unit/issues/41
//   * must include topic() + stateChangesOnly()

	weather
	| alert().id('Temperature')
		.topic('weather')
		.message('Temperature alert - batch')
		.warn(lambda: "mean" > 80)
		.crit(lambda: "mean" > 100)
		.stateChangesOnly()
    .log('/tmp/temperature_batch.tick.log')

// case with dynamic data

	snow
	| alert().id('Snow')
		.message('Snow alert - batch')
		.warn(lambda: "count" > 2)
		.crit(lambda: "count" > 5)
		.stateChangesOnly()
    .log('/tmp/snow_batch.tick.log')
I0326 03:30:26.250655     147 influxdb.go:95] DEBUG:: Influxdb added [snow,location=us-midwest precipitation=5   1585186217818311680] to http://localhost:8086/write?db=weather&rp=default
I0326 03:30:26.253084     147 influxdb.go:95] DEBUG:: Influxdb added [snow,location=us-midwest precipitation=10  1585186817818311680] to http://localhost:8086/write?db=weather&rp=default
I0326 03:30:26.254637     147 influxdb.go:95] DEBUG:: Influxdb added [snow,location=us-midwest precipitation=10  1585187417818311680] to http://localhost:8086/write?db=weather&rp=default
I0326 03:30:26.260366     147 influxdb.go:95] DEBUG:: Influxdb added [snow,location=us-midwest precipitation=10  1585188017818311680] to http://localhost:8086/write?db=weather&rp=default
I0326 03:30:26.261962     147 influxdb.go:95] DEBUG:: Influxdb added [snow,location=us-midwest precipitation=10  1585188617818311680] to http://localhost:8086/write?db=weather&rp=default
I0326 03:30:26.263748     147 influxdb.go:95] DEBUG:: Influxdb added [snow,location=us-midwest precipitation=10  1585189217818311680] to http://localhost:8086/write?db=weather&rp=default
I0326 03:30:26.265532     147 influxdb.go:95] DEBUG:: Influxdb added [snow,location=us-midwest precipitation=10  1585189817818311680] to http://localhost:8086/write?db=weather&rp=default
I0326 03:30:26.266927     147 influxdb.go:95] DEBUG:: Influxdb added [snow,location=us-midwest precipitation=10  1585190417818311680] to http://localhost:8086/write?db=weather&rp=default
Processing batch script alert_weather_batch.tick...
I0326 03:30:29.267116     147 kapacitor.go:149] DEBUG:: Kapacitor fetching status of: alert_weather_batch.tick
E0326 03:30:29.285754     147 test.go:137] DEBUG:: teardown test: Alert snow:: batch dynamic timestamps
I0326 03:30:29.286334     147 influxdb.go:119] DEGUB:: Influxdb delete monitor weather
I0326 03:30:29.286408     147 influxdb.go:147] DEGUB:: Influxdb checking database weather
E0326 03:30:29.288026     147 influxdb.go:165] DEBUG:: Database = weather
ts=2020-03-26T03:30:30.217Z lvl=error msg="error executing query" service=kapacitor task_master=main task=alert_weather_batch.tick node=query2 err="database not found: weather"
ts=2020-03-26T03:30:30.218Z lvl=error msg="error executing query" service=kapacitor task_master=main task=alert_weather_batch.tick node=query1 err="database not found: weather"
I0326 03:30:30.288275     147 influxdb.go:147] DEGUB:: Influxdb checking database weather
E0326 03:30:30.289786     147 influxdb.go:165] DEBUG:: Database = _internal
E0326 03:30:30.289900     147 influxdb.go:173] DEBUG:: No database = weather
E0326 03:30:30.289954     147 influxdb.go:127] DEGUB:: Database absent: weather
I0326 03:30:30.290011     147 influxdb.go:212] DEBUG:: Influxdb cleanup database q=DROP DATABASE "weather"
```

Not certain about repro rate.

Also, repro seems to be some dependent upon how users have influx and/or kapacitor set up. In this project, both are started externally via docker-compose, and I've had a moderate repro rate. However, in my work setup, we want to make sure the test container is self-sufficient and self-contained, so we start kapacitor and influx inside the container itself, before running kapacitor-unit. With that approach, the repro rate is much higher.

Which is to say, the crux of the problem might be delicate reliance upon response times / asynchronous behavior.

During debugging, I was able to confirm performing influx commands "DELETE DATABASE foo" then "SHOW DATABASES" in rapid succession would still show database foo briefly after the deletion. Additionally, the test teardown performs the database deletion first, and then the task deletion. It also appears shows the query error occurring AFTER the database deletion has begun -- though that part is more confusing, since the code seems clearly sequential; perhaps this is a quirk of error processing flow?

**Comment by DreadPirateShawn** [comment]
In the end, tightening the wiggly bits found above has resulted in fixing the repro rate completely.

Specifically,
* add monitor after database create/delete which monitors for `SHOW DATABASES` to reflect the change before proceeding
* move database deletion after the task deletion

The latter step was definitely necessary -- that's the last change I made to fix the repro. But the former step is (at worst) simply a good idea to ensure clean influx target states, and (at best) equally crucial to the fix, so I kept that as well.

Improved safety & teardown of databases is included in https://github.com/DreadPirateShawn/kapacitor-unit

(With apologies, I've fully forked rather than pushing upstream PRs, since the upstream doesn't really have an active maintainer.)
